src/dump.js

```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+const { encodeType } = require('./dump.complex')
+
+/**
+ * @param {any} val
+ * @param {Array<number>} ver
+ * @returns {Buffer}
+ */
+const dump = (val, ver = [4, 8]) => {
+  const symcache = []
```

Comment:
```js
if (!Buffer.isBuffer(val)) {
  val = Buffer.from(val)
}
```

Reply:
how about this?

Reply:
the value shouldn't be a buffer, it should by any nodejs value type, e.g. string, number, object...